src/TropicalGeometry/matrix.jl

```diff
@@ -29,15 +31,13 @@ julia> det(A)
 (5)
 ```
 """
-function det(A::Generic.MatSpaceElem{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
-    detA = zero(base_ring(A))
-    nrows(A)!=ncols(A) && return detA # return tropical zero if matrix not square
-    for sigma in AbstractAlgebra.SymmetricGroup(nrows(A)) # otherwise follow Leibniz Formula
-        detA += prod([ A[i,sigma[i]] for i in 1:nrows(A) ])
-    end
-    return detA
+function det(A::Generic.MatSpaceElem{<:TropicalSemiringElem})
```

Comment:
```suggestion
function det(A::MatrixElem{<:TropicalSemiringElem})
```
This should work for the abstract AA matrix type as well